experimental/Schemes/SpecOpen.jl

```diff
@@ -784,7 +784,9 @@ function restrict(
   end
   inc = inclusion_morphism(U, domain(f), check=check)
   help_map = compose(inc, f, check=check)
-  return SpecOpenMor(U, V, maps_on_patches(help_map), check=check)
+  Y = ambient(V)
+  h = [restrict(g, domain(g), Y, check=check) for g in maps_on_patches(help_map)]
```

Comment:
So `restrict` can also extend the codomain?

Reply:
No. But the maps on the patches of a `SpecOpenMor` are required to take values in the `ambient` of the codomain. And that needs to be assure when restricting.

Reply:
So it is kind of a hack?

Reply:
I wouldn't say so, really...

It is basically just how I constructed it for now. For any `SpecOpen` `U` which is covered by affine patches `U_i`, a  `SpecOpenMor` `f : U -> V \subset Y` is given by `SpecMor`s `f_i : U_i -> Y` which glue on the overlaps `U_i \cap U_j` and whose image is contained in `V`. 

It would just be way too restrictive to require `V` to be partitioned in a compatible way so that the `f_i` would take values in affine patches of `V`. So the requirement is that the codomain of `f_i` must be `Y`. Now when we restrict to closed subsets `A` in `U` and `B` in `V`, then the ambient scheme of `B` can not be `Y` anymore and we have to restrict the map on the affine patches accordingly.